mesonbuild/compilers/detect.py

```diff
@@ -453,7 +453,7 @@ def sanitize(p: str) -> str:
                 ccache + compiler, version, for_machine, is_cross, info,
                 exe_wrap, defines=defines, full_version=full_version, linker=linker)
 
-        if 'Intel(R) C++ Intel(R)' in err:
+        if 'Intel(R) C++ Intel(R)' in err or 'Intel(R) oneAPI DPC++/C++ Compiler for applications' in err:
             version = search_version(err)
             target = 'x86' if 'IA-32' in err else 'x86_64'
             cls = c.IntelClCCompiler if lang == 'c' else cpp.IntelClCPPCompiler
```

Comment:
So if I understand this correctly, we now support OneAPI on Windows by detecting it as `intel-cl`, which is... what? The predecessor of `intel-llvm` that is now deprecated and replaced by new unified branding?

Reply:
* classic compilers:
  * linux: intel
  * windows: intel-cl
 * oneapi compilers:
   * linux: intel-llvm
   * windows: intel-cl
   
I know for sure that linux compilers no longer accept some options and have adopted the llvm option names instead. The new id will make it possible to adapt behavior in build files.

I don't have any experience with the windows compilers. If the behavior has not changed, then keeping the same id would make it easier for users. I thought it would be possible to introduce a new name later if I find there are differences.

I can introduce a new id, ``intel-llvm-cl`` if you prefer.

Reply:
Well, really I guess I am asking whether you know if there are differences that need to be checked for. :D I have no idea whether there should be a new ID or not (I think I have mentioned before that I know next to nothing about Intel's compilers), but I would like to make sure that whatever we do, it's an intentional choice.

Thanks for clarifying.

Reply:
I just checked cmake. They use IntelLLVM on windows. I will change to intel-llvm-cl